Assets/Game/Scripts/Character/Health.cs

```diff
@@ -60,21 +60,21 @@ public void Hit(float damage, GameObject hitEffect, string hitSound)
       GetAnimator().SetBool(AnimationTrigger.dead, true);
       GetActor().GetCollider().enabled = false;
       GetActor().GetStatus().SetGameFinishStatus();
-      SpawnReward();
+      SpawnNewCharacter();
       _isDead = true;
     }
   }
 
-  private void SpawnReward()
+  private void SpawnNewCharacter()
   {
     var baseStats = gameObject.GetComponent<Actor>().GetBaseStats();
     if (baseStats.MinRewardFromCorpse <= baseStats.MaxRewardFromCorpse)
     {
       var rewardAmount = Random.Range(baseStats.MinRewardFromCorpse, baseStats.MaxRewardFromCorpse);
       if (rewardAmount > 0)
       {
-        GameObject reward = Instantiate(rewardAfterDeath, this.transform.position, Quaternion.identity);
-        reward.GetComponent<Reward>().Amount = rewardAmount;
+        GameObject NewCharacter = Instantiate(spawnAfterDeath, this.transform.position, Quaternion.identity);
```

Comment:
ShowNextLevel should have nothing to do with baseStats or reward.   You should do, the logic should be if the character has nextLevelCharacter prefab, then instantiate nextLevelCharacter.
private void ShowNextLevelCharacter()
{
   if (nextLevelCharacter != null) 
       Instantiate(nextLevelCharacter, this.transform.position, Quaternion.identity);
}